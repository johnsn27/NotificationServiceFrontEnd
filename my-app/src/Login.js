import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Btn from '@bbc/igm-btn';
import '@bbc/igm-btn/dist/Btn.css';
import '@bbc/igm-sticky-bar/dist/StickyBar.css';
import Test from './Test';

class Login extends Component {
  constructor(props) {
    super();
    this.toggleViewRoom = this.toggleViewRoom.bind(this);
    this.state = { DisplayViewRoom: props.DisplayViewRoom };
  }

  toggleViewRoom() {
    this.setState({ DisplayViewRoom: !this.state.DisplayViewRoom });
  }

  render() {
    return this.state.DisplayViewRoom
      ? <Test />
      : <div className="App">
        <header className="App-header"></header>
        <div className="loginButton">
          <Btn
            type="primary"
            tab-index="1"
            className="Button"
            onClick={this.toggleViewRoom}
          >
            <span>Login</span>
          </Btn>
        </div>
      </div >
  }
}

Login.propTypes = {
  DisplayViewRoom: PropTypes.bool,
};

Login.defaultProps = {
  DisplayViewRoom: false,
}

export default Login;
