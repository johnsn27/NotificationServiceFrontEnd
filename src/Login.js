import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Btn from '@bbc/igm-btn';
import '@bbc/igm-btn/dist/Btn.css';
import Main from './Main';

class Login extends Component {
  constructor(props) {
    super();
    this.toggleViewRoom = this.toggleViewRoom.bind(this);
    this.state = { DisplayViewRoom: props.DisplayViewRoom };
  }

  toggleViewRoom() {
    this.setState({ DisplayViewRoom: !this.state.DisplayViewRoom });
  }

  render() {
    return this.state.DisplayViewRoom
      ? <Main />
      : <div className="App">
        <header className="App-header"></header>
        <div className="loginButton">
          <Btn
            type="primary"
            tab-index="1"
            className="Button"
            onClick={this.toggleViewRoom}
          >
            <span>Login</span>
          </Btn>
        </div>
      </div >
  }
}

Login.propTypes = {
  DisplayViewRoom: PropTypes.bool,
};

Login.defaultProps = {
  DisplayViewRoom: false,
}

export default Login;
