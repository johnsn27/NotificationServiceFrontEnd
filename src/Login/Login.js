import React, { Component } from 'react';
import Btn from '@bbc/igm-btn';
import '@bbc/igm-btn/dist/Btn.css';

class Login extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header"></header>
        <div className="login-button">
          <Btn type="primary" tab-index="1" className="Button" onClick={() => window.location.href = '/home'}>
            <span>Login</span>
          </Btn>
        </div>
      </div >
    );
  }
}

export default Login;
