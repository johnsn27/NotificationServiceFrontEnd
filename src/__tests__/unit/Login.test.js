import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Login from '../../Login/Login';

describe('Login', () => {
  configure({ adapter: new Adapter() });

  it('renders without crashing', () => {
    shallow(<Login />);
  });

  it('matches snapshot', () => {
    const tree = shallow(<Login/>);
    expect(tree).toMatchSnapshot();
  })

  it('contains a button', () => {
  const app = shallow(<Login/>);
  expect(app.find('.Button').exists()).toBe(true);
  });
});
