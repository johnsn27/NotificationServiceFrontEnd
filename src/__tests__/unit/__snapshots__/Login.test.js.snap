// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    DisplayViewRoom={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
        />,
        <div
          className="loginButton"
        >
          <Btn
            className="Button"
            disabled={false}
            htmlType="button"
            onClick={[Function]}
            tab-index="1"
            type="primary"
          >
            <span>
              Login
            </span>
          </Btn>
        </div>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "App-header",
        },
        "ref": null,
        "rendered": null,
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Btn
            className="Button"
            disabled={false}
            htmlType="button"
            onClick={[Function]}
            tab-index="1"
            type="primary"
          >
            <span>
              Login
            </span>
          </Btn>,
          "className": "loginButton",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span>
              Login
            </span>,
            "className": "Button",
            "disabled": false,
            "htmlType": "button",
            "onClick": [Function],
            "tab-index": "1",
            "type": "primary",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
            },
            "ref": null,
            "rendered": "Login",
            "type": "span",
          },
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
          />,
          <div
            className="loginButton"
          >
            <Btn
              className="Button"
              disabled={false}
              htmlType="button"
              onClick={[Function]}
              tab-index="1"
              type="primary"
            >
              <span>
                Login
              </span>
            </Btn>
          </div>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "App-header",
          },
          "ref": null,
          "rendered": null,
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Btn
              className="Button"
              disabled={false}
              htmlType="button"
              onClick={[Function]}
              tab-index="1"
              type="primary"
            >
              <span>
                Login
              </span>
            </Btn>,
            "className": "loginButton",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <span>
                Login
              </span>,
              "className": "Button",
              "disabled": false,
              "htmlType": "button",
              "onClick": [Function],
              "tab-index": "1",
              "type": "primary",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
              },
              "ref": null,
              "rendered": "Login",
              "type": "span",
            },
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
